plotly_resampler/aggregation/aggregators.py

```diff
@@ -63,39 +82,189 @@ def _aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
             # default case, use the series it's dtype as return type
             data = data.astype(s.dtype)
 
-        return pd.Series(
-            index=s.iloc[idx.astype("uint32")].index.astype(s.index.dtype),
-            data=data,
-            name=str(s.name),
-            copy=False
+        return pd.Series(index=idx, data=data, name=str(s.name), copy=False)
+
+
+class MinMaxOverlapAggregator(AbstractSeriesAggregator):
+    """Aggregation method which performs binned min-max aggregation over 50% overlapping
+    windows.
+
+    .. image:: _static/minmax_operator.png
+
+    In the above image, **bin_size**: represents the size of *(len(series) / n_out)*.
+    As the windows have 50% overlap and are consecutive, the min & max values are
+    calculated on a windows with size (2x bin-size).
+
+    .. note::
+        This method is rather efficient when scaling to large data sizes and can be used
+        as a data-reduction step before feeding it to the :class:`LTTB <LTTB>`
+        algorithm, as :class:`EfficientLTTB <EfficientLTTB>` does.
+
+    """
+
+    def __init__(self, interleave_gaps: bool = True):
+        """
+        Parameters
+        ----------
+        interleave_gaps: bool, optional
+            Whether None values should be added when there are gaps / irregularly
+            sampled data. A quantile-based approach is used to determine the gaps /
+            irregularly sampled data. By default, True.
+        """
+        # this downsampler supports all pd.Series dtypes
+        super().__init__(interleave_gaps, dtype_regex_list=None)
+
+    def _aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
+        # The block size 2x the bin size we also perform the ceil-operation
+        # to ensure that the block_size =
+        block_size = math.ceil(s.shape[0] / (n_out + 1) * 2)
+        argmax_offset = block_size // 2
+
+        # Calculate the offset range which will be added to the argmin and argmax pos
+        offset = np.arange(
+            0, stop=s.shape[0] - block_size - argmax_offset, step=block_size
         )
 
+        # Calculate the argmin & argmax on the reshaped view of `s` &
+        # add the corresponding offset
+        argmin = (
+            s[: block_size * offset.shape[0]].values.reshape(-1, block_size).argmin(
+                axis=1)
+            + offset
+        )
+        argmax = (
+            s[argmax_offset: block_size * offset.shape[0] + argmax_offset]
+            .values.reshape(-1, block_size).argmax(axis=1)
+            + offset
+            + argmax_offset
+        )
+        # Sort the argmin & argmax (where we append the first and last index item)
+        # and then slice the original series on these indexes.
+        return s.iloc[np.unique(np.concatenate((argmin, argmax, [0, s.shape[0] - 1])))]
+
+
+class MinMaxAggregator(AbstractSeriesAggregator):
+    """Aggregation method which performs binned min-max aggregation over non-overlapping
```

Comment:
IMO these  are not non-overlapping but FULLY overlapping windows.
As the thing that overlaps is the min-window and max-window

Reply:
Agree, good catch! 🧐